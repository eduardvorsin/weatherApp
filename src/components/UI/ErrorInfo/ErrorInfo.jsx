import PropTypes from 'prop-types';
import { isCorrectHeadingLevel } from '../../../utils/utils';
import classes from './ErrorInfo.module.css';

export default function ErrorInfo({
  className,
  message,
  icon,
  testId,
  isCentered,
  headingLevel,
}) {
  const centerClass = isCentered ? `${classes['error-info--center']}` : '';
  const Heading = isCorrectHeadingLevel(headingLevel) ? `h${headingLevel}` : 'h6';

  return (
    <div
      className={`${classes['error-info']} ${className} ${centerClass}`}
      data-testid={testId}
      role="alert"
    >
      {icon && (
        <div
          className={classes['error-info__icon-wrapper']}
        >
          {icon}
        </div>
      )}

      <Heading
        className={classes['error-info__description']}
      >
        {message}
      </Heading>
    </div>
  );
}

ErrorInfo.defaultProps = {
  className: '',
  message: 'default message',
  icon: null,
  testId: '',
  isCentered: false,
  headingLevel: 6,
};

ErrorInfo.propTypes = {
  className: PropTypes.string,
  message: PropTypes.string,
  icon: PropTypes.element,
  testId: PropTypes.string,
  isCentered: PropTypes.bool,
  headingLevel: PropTypes.number,
};
