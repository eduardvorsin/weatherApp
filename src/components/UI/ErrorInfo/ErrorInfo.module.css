.error-info {
  padding: 15px;
  border: 3px solid var(--error-border-color);
  max-width: 750px;
  background-color: var(--error-bg-color);
  color: var(--error-color);
}

.error-info--center {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
}

.error-info__icon-wrapper {
  margin: 0 auto 15px;
  width: 60px;
  height: 60px;
}

.error-info__description {
  margin: 0;
  line-height: 1.3;
  text-align: center;
  font-size: 18px;
}

@media(min-width:577px) {
  .error-info {
    padding: 20px;
  }

  .error-info__description {
    font-size: 20px;
  }

  .error-info__icon-wrapper {
    width: 80px;
    height: 80px;
    margin: 0 auto 25px;
  }
}
