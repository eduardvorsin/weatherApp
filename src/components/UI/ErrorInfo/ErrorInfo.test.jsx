import { render, screen } from '@testing-library/react';
import ErrorInfo from './ErrorInfo';
import { ReactComponent as MockIcon } from '../../../assets/icons/settings.svg';

describe('ErrorInfo component tests', () => {
  it('renders correctly', () => {
    render(<ErrorInfo
      className="test-class"
      message="text"
      icon={<MockIcon />}
    />);

    expect(screen.getByText(/text/)).toBeInTheDocument();
  });

  it('renders without props', () => {
    render(<ErrorInfo />);

    expect(screen.getByText(/default message/)).toBeInTheDocument();
  });

  it('renders with icon', () => {
    render(<ErrorInfo
      message="text"
      icon={<MockIcon title="mock-icon" />}
    />);

    expect(screen.getByTitle(/mock-icon/)).toBeInTheDocument();
  });

  it('the classname is correctly assigned to the element', () => {
    render(<ErrorInfo
      className="test-class"
      message="text"
      testId="error-info"
    />);

    expect(screen.getByTestId('error-info')).toHaveClass('test-class');
  });

  it('when the correct headingLevel is passed, the header with the appropriate level is rendered inside the component', () => {
    render(<ErrorInfo
      className="test-class"
      message="text"
      testId="error-info"
      headingLevel={2}
    />);

    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
  });

  it('when an incorrect headingLevel is passed, a header with level 6 is rendered inside the component', () => {
    render(<ErrorInfo
      className="test-class"
      message="text"
      testId="error-info"
    />);

    expect(screen.getByRole('heading', { level: 6 })).toBeInTheDocument();
  });



  it('shapshot with default text', () => {
    render(<ErrorInfo
      testId="error-info"
    />);

    expect(screen.getByTestId('error-info')).toMatchSnapshot();
  });

  it('shapshot with mock text', () => {
    render(<ErrorInfo
      message="an important error"
      testId="error-info"
    />);

    expect(screen.getByTestId('error-info')).toMatchSnapshot();
  });

  it('shapshot with another heading level', () => {
    render(<ErrorInfo
      message="an important error"
      testId="error-info"
      headingLevel={4}
    />);

    expect(screen.getByTestId('error-info')).toMatchSnapshot();
  });

  it('shapshot with isCentered props', () => {
    render(<ErrorInfo
      message="an important error"
      testId="error-info"
      isCentered
    />);

    expect(screen.getByTestId('error-info')).toMatchSnapshot();
  });
});
