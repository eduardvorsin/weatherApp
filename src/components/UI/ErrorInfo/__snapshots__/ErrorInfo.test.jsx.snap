// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorInfo component tests shapshot with another heading level 1`] = `
<div
  class="error-info  "
  data-testid="error-info"
  role="alert"
>
  <h4
    class="error-info__description"
  >
    an important error
  </h4>
</div>
`;

exports[`ErrorInfo component tests shapshot with default text 1`] = `
<div
  class="error-info  "
  data-testid="error-info"
  role="alert"
>
  <h6
    class="error-info__description"
  >
    default message
  </h6>
</div>
`;

exports[`ErrorInfo component tests shapshot with isCentered props 1`] = `
<div
  class="error-info  error-info--center"
  data-testid="error-info"
  role="alert"
>
  <h6
    class="error-info__description"
  >
    an important error
  </h6>
</div>
`;

exports[`ErrorInfo component tests shapshot with mock text 1`] = `
<div
  class="error-info  "
  data-testid="error-info"
  role="alert"
>
  <h6
    class="error-info__description"
  >
    an important error
  </h6>
</div>
`;
